synthesizer/src/block/transaction/mod.rs

```diff
@@ -84,69 +84,91 @@ impl<N: Network> Transaction<N> {
         vm: &VM<N, C>,
         private_key: &PrivateKey<N>,
         program: &Program<N>,
-        (credits, fee_in_gates): (Record<N, Plaintext<N>>, u64),
+        (credits, fee_in_microcredits): (Record<N, Plaintext<N>>, u64),
         query: Option<Query<N, C::BlockStorage>>,
         rng: &mut R,
     ) -> Result<Self> {
         // Compute the deployment.
         let deployment = vm.deploy(program, rng)?;
         // Compute the fee.
-        let (_, fee, _) = vm.execute_fee(private_key, credits, fee_in_gates, query, rng)?;
+        let (_, fee, _) = vm.execute_fee(private_key, credits, fee_in_microcredits, query, rng)?;
         // Initialize the transaction.
         Self::from_deployment(deployment, fee)
     }
 
-    /// Initializes a new execution transaction from an authorization, and an optional fee.
-    pub fn execute_authorization<C: ConsensusStorage<N>, R: Rng + CryptoRng>(
+    /// Initializes a new execution transaction.
+    pub fn execute<C: ConsensusStorage<N>, R: Rng + CryptoRng>(
         vm: &VM<N, C>,
-        authorization: Authorization<N>,
+        private_key: &PrivateKey<N>,
+        (program_id, function_name): (impl TryInto<ProgramID<N>>, impl TryInto<Identifier<N>>),
+        inputs: impl ExactSizeIterator<Item = impl TryInto<Value<N>>>,
+        fee: Option<(Record<N, Plaintext<N>>, u64)>,
         query: Option<Query<N, C::BlockStorage>>,
         rng: &mut R,
     ) -> Result<Self> {
+        // Compute the authorization.
+        let authorization = vm.authorize(private_key, program_id, function_name, inputs, rng)?;
         // Compute the execution.
-        let (_response, execution, _metrics) = vm.execute(authorization, query, rng)?;
+        let (_response, execution, _metrics) = vm.execute(authorization, query.clone(), rng)?;
+        // Compute the fee.
+        let fee = match fee {
+            None => None,
+            Some((credits, fee_in_microcredits)) => {
+                Some(vm.execute_fee(private_key, credits, fee_in_microcredits, query, rng)?.1)
+            }
+        };
         // Initialize the transaction.
-        Self::from_execution(execution, None)
+        Self::from_execution(execution, fee)
     }
 
-    /// Initializes a new execution transaction from an authorization, and an optional fee.
-    pub fn execute_authorization_with_additional_fee<C: ConsensusStorage<N>, R: Rng + CryptoRng>(
+    /// Initializes a new fee.
+    pub fn execute_fee<C: ConsensusStorage<N>, R: Rng + CryptoRng>(
         vm: &VM<N, C>,
         private_key: &PrivateKey<N>,
-        authorization: Authorization<N>,
-        additional_fee: Option<(Record<N, Plaintext<N>>, u64)>,
+        credits: Record<N, Plaintext<N>>,
+        fee_in_microcredits: u64,
```

Comment:
For clarity can it be explained how credits are going to be associated with records in general? I think the explanation for this should probably go in the top level documentation for the method.

Also naming-wise, this method is a little opaque on first look. I believe this is saying "the credits associated with this record are being used to pay the fee" 

Thus might be something like the following:
```suggestion
        unspent_funds: Record<N, Plaintext<N>>,
        fee_in_microcredits: u64,
```